here-naksha-app-service/src/main/java/com/here/naksha/app/service/http/tasks/AbstractApiTask.java

```diff
@@ -222,4 +241,29 @@ protected Result executeWriteRequestFromSpaceStorage(WriteFeatures writeRequest)
   private XyzFeatureCollection emptyFeatureCollection() {
     return new XyzFeatureCollection().withFeatures(emptyList());
   }
+
+  private Map<String, Object> removePasswordFromProps(Map<String, Object> propertiesAsMap) {
+    for (Iterator<Map.Entry<String, Object>> it = propertiesAsMap.entrySet().iterator(); it.hasNext(); ) {
+      Map.Entry<String, Object> entry = it.next();
+      if (Objects.equals(entry.getKey(), "password")) {
+        it.remove();
+      } else if (entry.getValue() instanceof Map) {
+        // recursive call to the nested json property
+        removePasswordFromProps((Map<String, Object>) entry.getValue());
+      } else if (entry.getValue() instanceof ArrayList array) {
+        // recursive call to the nested array json
+        for (Object arrayEntry : array) {
+          removePasswordFromProps((Map<String, Object>) arrayEntry);
+        }
+        entry.setValue(array);
+      }
+    }
+    return propertiesAsMap;
+  }
+
+  private <R extends XyzFeature> void removePasswordFromFeature(final @NotNull R feature) {
+    Map<String, Object> propertiesAsMap =
+        removePasswordFromProps(feature.getProperties().asMap());
+    feature.setProperties(JsonSerializable.fromMap(propertiesAsMap, XyzProperties.class));
```

Comment:
I don't think we have to set it again. It should work without this line as well. I suggest trying it out.